core/services/vrf/v2/listener_v2_log_listener.go

```diff
@@ -0,0 +1,451 @@
+package v2
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+	"github.com/smartcontractkit/chainlink/v2/core/services/vrf/vrfcommon"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+	"github.com/smartcontractkit/chainlink/v2/core/utils/mathutil"
+)
+
+func (lsn *listenerV2) runLogListener(
+	pollPeriod time.Duration,
+	minConfs uint32,
+) {
+	lsn.l.Infow("Listening for run requests via log poller",
+		"minConfs", minConfs)
+	ticker := time.NewTicker(utils.WithJitter(pollPeriod / 2))
+	defer ticker.Stop()
+	var (
+		lastProcessedBlock int64
+		startingUp         = true
+	)
+	ctx, cancel := lsn.chStop.NewCtx()
+	defer cancel()
+	for {
+		select {
+		case <-lsn.chStop:
+			return
+		case <-ticker.C:
+			start := time.Now()
+			lsn.l.Debugw("log listener loop")
+			// Filter registration is idempotent, so we can just call it every time
```

Comment:
i'm confused by this. if it's idempotent, then if you do it once successfully, isn't that enough? this seems to incur extra processing on every tick that has no utility

Reply:
The reason I have it here in the loop is to retry if it fails. I'm not sure under which scenario it can fail though